KitbotExampleWalkthrough.md

```diff
@@ -0,0 +1,245 @@
+
+We might not want to drive the robot just by setting the left and right wheel speeds, however.
+Lets add a method that uses a desired forwards/backwards and turning velocity to set the wheel speeds.
+We can do this using WPILib's `DifferentialDrive` class.
+
+```Java
+public CommandBase setVoltagesArcadeCommand(DoubleSupplier drive, DoubleSupplier steer) {
+    return new RunCommand(() -> {
+      var speeds = DifferentialDrive.arcadeDriveIK(drive.getAsDouble(), steer.getAsDouble(), false);
+      this.setVoltages(speeds.left * 12, speeds.right * 12);
+    }, this);
+}
+```
+
+This method is named `setVoltagesArcadeCommand` after the arcade drive inverse kinematics (IK) it uses, but other types of IK can also work.
+On a real robot what we use would come down to driver preference.
+In the method itself we have a very similar structure to the previous method.
+The main difference is that instead of having one method in our lambda (`() ->`), we have a code block ({} block).
+This means we need to end each line with a ; but can use multiple lines in our Command.
+In this case we call arcadeDriveIK from WPILib's `DifferentialDrive` class and store the result in a variable called `speeds`.
+Then we pass speeds to the drive method, multiplying by 12 to convert from the -1..1 units from the IK method to voltage.
+
+These methods are all well and good on their own, but it would be nice if we had a way to actually use them.
+Lets bind them to a joystick.
+Go to RobotContainer.java and add a `CommandXboxController` object.
+
+```Java
+CommandXboxController controller = new CommandXboxController(0);
+```
+
+`CommandXboxController` is a convenience wrapper around a `XboxController` that makes it easy to bind commands to buttons on the controller.
+We also need to add an instance of the `DrivetrainSubsystem` to RobotContainer so that we can control it.
+
+```Java
+CommandXboxController controller = new CommandXboxController(0);
+
+DrivetrainSubsystem drivetrainSubsystem = new DrivetrainSubsystem();
+```
+
+Finally we can bind the arcade drive command to the joysticks.
+To do this call `setDefaultCommand()` on `drivetrainSubsystem` in RobotContainer's `configureBindings()` method.
+This sets a Command that will be run when no other command is using the drivetrain.
+Pass in `drivetrainSubsystem.setVoltagesArcadeCommand()` as the default Command.
+
+For the `DoubleSuppliers` to the arcade drive command we can add two lambdas that call `getLeftY()` and `getRightX()` on the controller, which looks at the current joysick values.
+This means that each loop this command is run, it will check the current joystick values and turn them into voltages for the wheels.
+
+This line is a little long now, so break it up into a few lines to keep it readable.
+
+```Java
+drivetrainSubsystem.setDefaultCommand(
+    drivetrainSubsystem.setVoltagesArcadeCommand(
+        () -> controller.getLeftY(), 
+        () -> controller.getRightX()));
```

Comment:
You can use a [method reference](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html): `controller::getLeftY` and `controller::getRightX` and skip the extra lambda syntax.

`drivetrainSubsystem.setVoltagesArcadeCommand(controller::getLeftY, controller::getRightX)`

Reply:
I avoided that since its more new syntax and because I'm going to wrap these in a method call later.